repositories/decisions_repository.go

```diff
@@ -434,15 +434,27 @@ func (repo *DecisionRepositoryImpl) rulesOfDecision(ctx context.Context, exec Ex
 		return nil, err
 	}
 
-	return SqlToListOfModels(
+	var columns []string
+	columns = append(columns, columnsNames("dr", dbmodels.SelectDecisionRuleColumn)...)
+	columns = append(columns, columnsNames("r", dbmodels.SelectDecisionRuleDetailColumn)...)
+	query := NewQueryBuilder().
+		Select(columns...).
+		From(fmt.Sprintf("%s AS dr", dbmodels.TABLE_DECISION_RULE)).
+		LeftJoin(fmt.Sprintf("%s AS r ON r.id = dr.rule_id", dbmodels.TABLE_RULES)).
+		Where(squirrel.Eq{"dr.decision_id": decisionId}).
+		OrderBy("dr.id")
+
+	return SqlToListOfRow(
 		ctx,
 		exec,
-		NewQueryBuilder().Select(dbmodels.SelectDecisionRuleColumn...).
-			From(dbmodels.TABLE_DECISION_RULE).
-			Where(squirrel.Eq{"decision_id": decisionId}).
-			OrderBy("id"),
-		func(r dbmodels.DbDecisionRule) (models.RuleExecution, error) {
-			return dbmodels.AdaptRuleExecution(r), nil
+		query,
+		func(row pgx.CollectableRow) (models.RuleExecution, error) {
+			db, err := pgx.RowToStructByPos[dbmodels.DbDecisionRuleWithDetail](row)
```

Comment:
I think I'm a bit troubled by this utility function, in this context.
I fail to understand how it reconciles tables and fields. You have prefixes on fields and table names in the query (that's good), but the prefixes are of course not present on the (composed) struct attribute names. Especially here where the two joined tables share some field name score_modifier (though not the nested structs).

So by default (assuming you tested it and it worked locally) it does just take them in order, and understands that the first one goes to the DbDecisionRule and the second to the DbDecisionRuleDetail. (what happens if you invert the order in the columns by switching lines 438 and 439 btw ?)

Anyway, this is where pgx.RowToStructByPos becomes hard to follow and we're perhaps better off by explicitly scanning into the struct.